libs/agno/agno/agent/agent.py

```diff
@@ -4072,9 +4159,12 @@ def _format_message_with_state_variables(
         if not isinstance(message, str):
             return message
 
+        # Use provided dependencies or fall back to agent dependencies
+        effective_dependencies = dependencies if dependencies is not None else self.dependencies
```

Comment:
Same here, rather figure it out in `run()`. That is the general pattern, figure out the variable "for the current run" and pass it down.